cmd/geth/version_check_test.go

```diff
@@ -64,16 +68,63 @@ func testVerification(t *testing.T, pubkey, sigdir string) {
 	}
 }
 
-func TestJson(t *testing.T) {
-	data, _ := ioutil.ReadFile("./testdata/vcheck/data2.json")
+func versionUint(v string) int {
+	mustInt := func(s string) int {
+		a, err := strconv.Atoi(s)
+		if err != nil {
+			panic(v)
+		}
+		return a
+	}
+	components := strings.Split(strings.TrimPrefix(v, "v"), ".")
+	a := mustInt(components[0])
+	b := mustInt(components[1])
+	c := mustInt(components[2])
+	return a*100*100 + b*100 + c
+}
+
+// TestMatching can be used to check that the regexps are correct
+func TestMatching(t *testing.T) {
+	data, _ := ioutil.ReadFile("./testdata/vcheck/vulnerabilities.json")
 	var vulns []vulnJson
 	if err := json.Unmarshal(data, &vulns); err != nil {
 		t.Fatal(err)
 	}
-	if len(vulns) == 0 {
-		t.Fatal("expected data, got none")
+	check := func(version string) {
+		vFull := fmt.Sprintf("Geth/%v-unstable-15339cf1-20201204/linux-amd64/go1.15.4", version)
+		for _, vuln := range vulns {
+			r, err := regexp.Compile(vuln.Check)
+			vulnIntro := versionUint(vuln.Introduced)
+			vulnFixed := versionUint(vuln.Fixed)
+			current := versionUint(version)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if vuln.Name == "Denial of service due to Go CVE-2020-28362" {
```

Comment:
thats a bit weird

Reply:
Yeah, I know... I'll try to figure out something better, but it's a bit quirky since the bug isn't tied to an actual geth-version